src/MongoDB.Driver.Core/Core/Servers/ServerMonitor.cs

```diff
@@ -52,7 +54,26 @@ internal sealed class ServerMonitor : IServerMonitor
         public event EventHandler<ServerDescriptionChangedEventArgs> DescriptionChanged;
 
         public ServerMonitor(ServerId serverId, EndPoint endPoint, IConnectionFactory connectionFactory, TimeSpan heartbeatInterval, TimeSpan timeout, IEventSubscriber eventSubscriber)
+            : this(serverId, endPoint, connectionFactory, heartbeatInterval, timeout, eventSubscriber, new CancellationTokenSource())
         {
+        }
+
+        public ServerMonitor(ServerId serverId, EndPoint endPoint, IConnectionFactory connectionFactory, TimeSpan heartbeatInterval, TimeSpan timeout, IEventSubscriber eventSubscriber, CancellationTokenSource cancellationTokenSource)
```

Comment:
added this to be able to use DI for some of ctor arguments

Reply:
Love DI :)  ╰(*°▽°*)╯╰(*°▽°*)╯